examples/webhook_ping_pong_bot/src/main.rs

```diff
@@ -0,0 +1,85 @@
+// The version of ping-pong-bot, which uses a webhook to receive updates from
+// Telegram, instead of long polling.
+
+use teloxide::{dispatching::update_listeners, prelude::*};
+
+use std::{convert::Infallible, net::SocketAddr, sync::Arc};
+use tokio::sync::mpsc;
+use warp::Filter;
+
+use reqwest::StatusCode;
+
+#[tokio::main]
+async fn main() {
+    run().await;
+}
+
+async fn handle_rejection(
+    error: warp::Rejection,
+) -> Result<impl warp::Reply, Infallible> {
+    log::error!("Cannot process the request due to: {:?}", error);
+    Ok(StatusCode::INTERNAL_SERVER_ERROR)
+}
+
+pub async fn webhook<'a>(
+    bot: Arc<Bot>,
+) -> impl update_listeners::UpdateListener<Infallible> {
+    // You might want to specify a self-signed certificate via .certificate
+    // method on SetWebhook.
+    bot.set_webhook("Your HTTPS ngrok URL here. Get it by 'ngrok http 80'")
+        .send()
+        .await
+        .expect("Cannot setup a webhook");
+
+    let (tx, rx) = mpsc::unbounded_channel();
+
+    let server = warp::post()
+        .and(warp::body::json())
+        .map(move |json: serde_json::Value| {
+            match serde_json::from_str::<Update>(&json.to_string()) {
```

Comment:
How about create function `parse_update(value: serde_json::Value) -> Option<Update>` in utils module?

Reply:
I think we need to provide the same behaviour for `from_str` and `from_value` instead of making such a function.

Link: https://github.com/serde-rs/json/issues/618